### PR TITLE
Add beta feature to prevent deleting payment methods with subscriptions

### LINKED ISSUE #448: vaultManager prevent deleting subscription payment method
Drop-in UI js v3 SDK

It would be nice to have some more options related to the vaultManager setting. For example, we want customers to be able to edit/remove saved payment methods, except for ones tied to a subscription. Maybe another setting to prevent that case? Or just updating the current behavior to prevent removing those payment methods? 

Thanks!

### PR DESCRIPTION
**Pull request by crookedneighbor**
closes #448

### Summary
Marked the feature as beta so we're free to iterate on the design and decide if we want to add messaging about why the payment method is disabled.

prevent-deleting-subscription

### Checklist

- [x] Added a changelog entry

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,9 @@ Unreleased
 - Clean up extraneous css rules
 - Add 3DS 2 support
 
+_Beta Features_
+- Add ability to prevednt deleting a payment method with an associated subscription
```
Comment (cooperbarth) [inline CHANGELOG.md]:
```suggestion
- Add ability to prevent deleting a payment method with an associated subscription
```

**Comment by AndreiChe** [comment]
Please merge. Really need it!

**Comment by raRaRa** [comment]
Is this feature live? Thanks.

**Comment by riteshkhatri** [comment]
Is this change merged to master? i realy need it.

**Comment by crookedneighbor** [comment]
This still needs some significant design work to be able to be merged.

**Comment by jstackman** [comment]
I haven't looked at the code of this at all.  What works still needs to be done?  I might be able to help.

**Comment by crookedneighbor** [comment]
There's some design considerations that still need to be worked out, but most of it is that other priorities have come up, so this has been pushed back.

**Comment by ItsReddi** [comment]
Can you provide an eta for this to be able to plan our views?
Something like: This year - Next year - in 10 years - never
Would be enough.
Its hanging around almost 2 years an our developers rechecking that again and again

Thanks

**Comment by crookedneighbor** [comment]
I don't have an eta except to say that this is not a high priority feature. There are many other things we have in the works that take precedence over this.

As such, I'm going to close this PR rather than leave it open. We can revisit it later on.

**Pull request closed without merging**